pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1277,6 +1288,9 @@ async function resolveDependency (
     if (!options.update && currentPkg.version && currentPkg.pkgId?.endsWith(`@${currentPkg.version}`) && !calcSpecifier) {
       wantedDependency.bareSpecifier = replaceVersionInBareSpecifier(wantedDependency.bareSpecifier, currentPkg.version)
     }
+    if (!options.updateRequested && options.preferredVersion !== undefined) {
```

Comment:
I'd check for `!= null` to cover both undefined and null.

Reply:
Good idea!